MapboxCoreNavigation/RouteController.swift

```diff
@@ -635,7 +561,8 @@ extension RouteController: CLLocationManagerDelegate {
             currentStepProgress.distanceTraveled = distanceTraveled
             NotificationCenter.default.post(name: .routeControllerProgressDidChange, object: self, userInfo: [
                 RouteControllerNotificationUserInfoKey.routeProgressKey: routeProgress,
-                RouteControllerNotificationUserInfoKey.locationKey: location
+                RouteControllerNotificationUserInfoKey.locationKey: self.location!, // Idealized location, raw is guaranteed
```

Comment:
Unwrap if you can.

Reply:
The reason why the force-unwrap is used here is because the raw location is guaranteed.

```swift
guard let location = potentialLocation else { return }
         
self.rawLocation = location
```
[Link](https://github.com/mapbox/mapbox-navigation-ios/pull/1188/files/797437a4705bd129245d706acb8f77da16a8f3cb#diff-1a2149692b3d7c58b6a0392120d6011aR541)